modules/serverless-build-func-kn.adoc

```diff
@@ -11,13 +11,13 @@ Before you can run a function, you must build the function project by using the
 .Example `func.yaml`
 [source,yaml]
 ----
-name: example-function
-namespace: default
-runtime: node
+name: <function_name>
+namespace: <function_namespace>
+runtime: <function_runtime>
 image: <image_from_registry>
 imageDigest: ""
 trigger: http
-builder: default
+builder: <builder_type>
```

Comment:
I am afraid `builder` is not a builder-type, it suspect it's container image.
However the CLI flag `--builder` is in fact builder-type: `s2i` or `pack`.

Reply:
ACK, it's very confusing that they're names the same thing. Why not change this to "container-image" in that case instead of "builder" in the YAML?

Reply:
This is called "Builder" because it was created before there was such thing as differing builder types (buildpack and s2i).
It is most accurately "The image to use by the builder when building the Function."
So there is a Builder (S2I, Buildpack) hence the flag `--builder` and the image the builder uses to build (a container image) hence `--builder-image`.
As you can see, these are appropriately named.  However, in acknowlidging this can seem confusing on first glance, I have opened an issue where we can discuss better options.
One suggestion was to use "strategy" and "builder".
I still feel that "func build --builder=s2i --build-imag=quay.io/me/my-custom-builder-image" is the most accurate; and we just need to update func.yaml to be in sync; buit I am open to suggestions